src/model/dto/BucketDto.ts

```diff
@@ -16,7 +16,7 @@ export class BucketDto {
     public files: FileUploadResponseDto[];
 
     public static fromModel(model: BucketModel, baseUrl: string): BucketDto {
-        const fileDtos = model.files.map(f => FileUploadResponseDto.fromModel(f, baseUrl));
+        const fileDtos = model.files ? model.files.map(f => FileUploadResponseDto.fromModel(f, baseUrl)) : [];
```

Comment:
why? if tghe bucket is empty, the files array is empty. so why are you checking if files is nullish? can fileds even be nullish?